File: factors.c
#include <stdio.h>
#include <stdlib.h>
#include <stdint.h>
#include <unistd.h>
#include <ctype.h>
#include <math.h>
#include <inttypes.h> /* for PRIu64*/
/**
 * main - Factorize as many numbers as possible into
 * a product of two smaller numbers
 * @argc: command line arguments count
 * @argv: commmand line arguments vector
 * Return: void
 */
int main(int argc, char *argv[])
{
	FILE *file;
	char buffer[1024];
	uint64_t i, num;

	if (argc != 2)
	{
		dprintf(STDERR_FILENO, "Usage: factors <file>\n");
		exit(EXIT_FAILURE);
	}
	file = fopen(argv[1], "r");
	if (file == NULL)
	{
		perror("Error opening file");
		exit(EXIT_FAILURE);
	}
	while (fgets(buffer, sizeof(buffer), file) != NULL)
	{
		num = strtoull(buffer, NULL, 10);
		for (i = 2; i <= (uint64_t)sqrt(num); i++)
		{
			if (num % i == 0)
			{
				dprintf(STDOUT_FILENO, "%" PRIu64 "=%" PRIu64 "*%" PRIu64 "\n", num, num / i, i);
				break;
			}
		}

	}
	fclose(file);
	return (0);
}

File: factors.py
#!/usr/bin/python3
"""Import sys to access sys functions like argv, exit, etc."""
import sys
"""Import math to access math function like sqrt"""
import math


def factors():
    """Factorize a number into 2 smaller numbers"""
    if len(sys.argv) != 2:
        print("Usage: factors <file>")
        sys.exit(1)
    try:
        with open(sys.argv[1], "r") as numfile:
            for num in numfile:
                n = int(num.strip())
                for i in range(2, math.isqrt(num) + 1):
                    if n % i == 0:
                        print("{}={}*{}".format(n, n // i, i))
                        break
    except FileNotFoundError:
        print("File not found")
        sys.exit(1)

if __name__ == "__main__":
    factors()
